client/my-sites/promote-post-i2/components/posts-table/index.tsx

```diff
@@ -6,24 +7,30 @@ import PostsListHeader from '../posts-list/header';
 
 interface Props {
 	posts: BlazablePost[];
+	type: string;
 	isLoading: boolean;
 	isFetchingPageResults: boolean;
 }
 
 export default function PostsTable( props: Props ) {
-	const { posts, isLoading, isFetchingPageResults } = props;
+	const { posts, type, isLoading, isFetchingPageResults } = props;
+	const isRunningInWooStore = config.isEnabled( 'is_running_in_woo_site' );
 
 	return (
 		<table className="promote-post-i2__table">
-			<PostsListHeader />
+			{ isRunningInWooStore && type === 'product' ? (
```

Comment:
We can change this ternary if, and move it inside the postType prop, like this:
```
<PostsListHeader postType={ isRunningInWooStore && type === 'product' ? type : 'post } />
```

This way, we only mount the `PostsListHeader` component once, and just change the prop when needed. This will be more efficient in React.

Reply:
Fixed